content/src/service/access/AccessCheckerForWearables.ts

```diff
@@ -59,33 +71,61 @@ export class AccessCheckerForWearables {
   }
 
   private async checkCollectionAccess(
-    subgraphUrl: string,
+    blocksSubgraphUrl: string,
+    collectionsSubgraphUrl: string,
     collection: string,
     itemId: string,
+    timestamp: Timestamp,
     ethAddress: EthAddress
   ): Promise<boolean> {
     try {
-      const ethAddressLowercase = ethAddress.toLowerCase()
-      const permissions: WearableItemPermissionsData = await this.getCollectionItems(subgraphUrl, collection, itemId)
+      const { blockNumberNow, blockNumberFiveMinBefore } = await this.findBlocksForTimestamp(
+        blocksSubgraphUrl,
+        timestamp
+      )
+      // It could happen that the subgraph hasn't synced yet, so someone who just lost access still managed to make a deployment. The problem would be that when other catalysts perform
+      // the same check, the subgraph might have synced and the deployment is no longer valid. So, in order to prevent inconsistencies between catalysts, we will allow all deployments that
+      // have access now, or had access 5 minutes ago.
       return (
-        (permissions.collectionCreator && permissions.collectionCreator === ethAddressLowercase) ||
-        (permissions.collectionManagers && permissions.collectionManagers.includes(ethAddressLowercase)) ||
-        (permissions.itemManagers && permissions.itemManagers.includes(ethAddressLowercase))
+        (await this.hasPermission(ethAddress, collectionsSubgraphUrl, collection, itemId, blockNumberNow)) ||
```

Comment:
is this `blockNumberNow` at current time or at deployment time?

Reply:
Good point! It's at the deployment's time, but it definitely isn't clear by its name